### PR TITLE
LibLine: Bracketed paste support / Shell+LibLine: Allow some programs to modify the default termios

### LINKED ISSUE #7276: Terminal: Drag-and-drop file name containing new lines (CRLF) into Terminal executes each line as a command
image

### PR DESCRIPTION
**Pull request by alimpfard**
- LibLine: Add bracketed paste mode support 

Fixes #7276.

-  LibLine+Shell: Allow some programs to modify the current termios 

cc @BertalanD - you requested this.

### CODE REVIEW

**Comment by BertalanD** [comment]
Thank you for doing this.

I'm not sure if this is worth looking into, but shouldn't we match against the full `/bin/stty` path instead? If I understand this code correctly, any binary that's in PATH and is named `stty` can change these settings.

**Comment by alimpfard** [comment]
Any program that is run by invoking `stty` can, to be precise.
The shell doesn't resolve program names, so it can't really know if `stty` is a script in your PATH, or `/bin/stty`.

**Comment by alimpfard** [comment]
I don't think that's a cause of concern, this is all explicit user input, "allow termios modifications when I run a binary as `stty`"

**Comment by alimpfard** [comment]
Update: Addressed https://github.com/SerenityOS/serenity/pull/7389#issuecomment-847041151

Now LibLine will reevaluate the key bindings (correctly) when a program causes a change.

File: Userland/Libraries/LibLine/Editor.h [hunk 1]
```diff
@@ -89,6 +98,7 @@ struct Configuration {
     OperationMode operation_mode { OperationMode::Unset };
     Vector<KeyBinding> keybindings;
     String m_default_text_editor {};
+    bool enable_bracketed_paste { false };
```
Comment (linusg) [inline Userland/Libraries/LibLine/Editor.h:101]: There's a confusing mix of `m_` and no `m_` going on in this struct, but ok.

**Review by linusg — APPROVED** [review]
LGTM!

**Pull request merged**